scripts/add_directory.js

```diff
@@ -0,0 +1,287 @@
+                            <span class="col-sm-6 col-xs-12">
+                                <button id="batch-submit-${num}" class="btn btn-success" onclick="submitBatch(${num})"><span class="hidden rotating glyphicon glyphicon-repeat"></span>Create</button>
+                            </span>
+                            <span class="col-sm-6 col-xs-12">
+                                <button id="batch-cancel-${num}" class="btn btn-danger" onclick="deleteBatch(${num})">Cancel</button>
+                            </span>
+                        </div>
+                    </span>
+                </div>
+                <div class="alert-wrapper" id="alert-wrapper-${num}" class="row"></div>
+            </div>
+        </div>`;
+}
+
+function submitBatch(n) {
+    let batchReq = {
+        dir: $("#name-input-" + n).val(),
+        recursive: $("#recursive-check-" + n)[0].checked,
+        batch_name: $("#batch-dir-" + n).text()
+    };
+    let buttons = $(`[data-batch=${n}] button`);
+    buttons.toggleClass("disabled");
+    $(".glyphicon", buttons).toggleClass("hidden");
+    // eventually this will make an actual post request
+    setTimeout(function() {
+        res = [{result: "success", err_msg: null, img_errs: []},
+            {result: "error", err_msg: "Unhashable images detected", img_errs: ["02_98.png", "03_98.png"]},
+            {result: "fail", err_msg: "Database unreachable", img_errs: []}
+            ][Math.floor(Math.random() * 100) % 3];
+        let resAlert = "<div class='col-xs-10 col-xs-offset-1 alert ";
+        switch (res.result) {
+            case "success":
+                resAlert += "alert-success'><strong>Success:</strong> Batch Created</div>";
+                buttons.each(function(i, b) { b.onclick = undefined;});
+                break;
+            case "error":
+                resAlert += `alert-warning'><strong>Success:</strong> Batch created, some images skipped. Errors hashing ${res.img_errs.join(", ")}.</div>`;
+                buttons.each(function(i, b) { b.onclick = undefined;});
+                break;
+            case "fail":
+                resAlert += `alert-danger'><strong>Failed: </strong>${res.err_msg}</div>`;
+                $(buttons[0]).text("Retry");
+                buttons.toggleClass("disabled");
+                break;
+            default:
+                console.log("Unexpected response");
+                break;
+        }
+        $("#alert-wrapper-" + n).html($(resAlert));
+        $(".glyphicon", buttons).toggleClass("hidden");
+    }, 1000)
+}
+
+function deleteBatch(n) {
+    $(`[data-batch=${n}]`).fadeOut("slow", function() {
+        batchDirs.delete($("h3", this).text());
+        $(this).remove();
+    });
+}
+
+function addSubfolders() {
+    let parentDir = $("#folder-input").val();
+    let filterReg = new RegExp(`^${parentDir}\\w+\/?$`);
+    let res = autocomplete._list.filter(filterReg.test.bind(filterReg)).map(addBatch);

```

Comment:
it searches the list in autocomplete (which is all of our folders) runs them through the regex filter there (which looks for all directories that are direct children of parentDir) and then makes a batch out of all of them